### PR TITLE
BaSP-TG-69: Add Update Projects

### PR DESCRIPTION
**Pull request by maxidelfino**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "web-vitals": "^1.1.2"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "react-scripts --openssl-legacy-provider start",
```
Comment (baudraccoverona) [inline package.json:19]: remove this line

File: src/Components/Projects/Form/index.js [hunk 2]
```diff
@@ -144,86 +177,39 @@ const ProjectForm = () => {
...
-          setEmployees(content.data);
-        } else {
-          setServerError(content.message);
-          setShowModal(true);
-        }
-      })
-      .catch((error) => alert(error));
-  }, []);
-
+  if (loadProjects || loadEmployee) {
```
Comment (baudraccoverona) [inline src/Components/Projects/Form/index.js:206]:
```suggestion
  if (projectsIsLoading || employeesIsLoading) {
```

File: src/redux/employees/thunks.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      .then((res) => res.json())
+      .then((res) => {
+        if (res.error) {
+          throw new Error(res.message);
+        } else {
+          dispatch(getEmployeesSuccess(res.data));
+        }
+      })
+      .catch((err) => {
+        dispatch(getEmployeesError(err.toString()));
```
Comment (baudraccoverona) [inline src/redux/employees/thunks.js:16]:
```suggestion
        return dispatch(getEmployeesError(err.toString()));
```

File: src/redux/projects/thunks.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      .then((res) => res.json())
+      .then((res) => {
+        if (res.error) {
+          throw new Error(res.message);
+        } else {
+          dispatch(getProjectSuccess(res.data));
+        }
+      })
+      .catch((err) => {
+        dispatch(getProjectError(err.toString()));
```
Comment (baudraccoverona) [inline src/redux/projects/thunks.js:29]:
```suggestion
        return dispatch(getProjectError(err.toString()));
```

File: src/Components/Projects/Form/index.js [hunk 5]
```diff
@@ -144,86 +177,39 @@ const ProjectForm = () => {
...
+    return (
+      <>
```
Comment (baudraccoverona) [inline src/Components/Projects/Form/index.js:208]: remove this

File: src/Components/Projects/Form/index.js [hunk 6]
```diff
@@ -144,86 +177,39 @@ const ProjectForm = () => {
...
+        <Spinner isLoading={true} />
+      </>
```
Comment (baudraccoverona) [inline src/Components/Projects/Form/index.js:210]: remove this

**Review by lmalvasia — DISMISSED** [review]

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**